cpp-terminal/event.hpp

```diff
@@ -21,25 +22,33 @@ class Event
     Cursor,
     CopyPaste,
   };
-  Event() = default;
+  Event();
   Event(const std::string&);
   Event(const Term::Key&);
   Event(const Term::Screen& screen);
-  bool empty() const;
-  Type type() const;
+  Event(const Term::Cursor& cursor);
+  bool         empty() const;
+  Type         type() const;
+  Term::Event& operator=(const Term::Event& event) noexcept;
+  Event(const Term::Event& event) noexcept;
+  Event(Term::Event&& event) noexcept;
+  Event& operator=(Event&& other) noexcept;
 
   operator Term::Key() const;
   operator Term::Screen() const;
   operator Term::Cursor() const;
   operator std::string() const;
 
 private:
-  void         parse();
-  Type         m_Type{Type::Empty};
-  std::string  m_str;
-  Key          m_Key{Key::Value::NO_KEY};
-  Cursor       m_Cursor;
-  Term::Screen m_Screen;
+  void parse();
+  Type m_Type;
+  union
```

Comment:
I believe unnamed unions are not ISO C++. However, some compilers allow it, but maybe this generates some warnings?

Reply:
Oh, I thought it was in the standard. I can find a name for it if needed

Reply:
added a name and I redesigned parse to maybe avoid useless assignement of string